create-rust-functions

vrun1208 opened the pull request:
#3371 
### Description

Wrote about rust functions definition, declaration followed with various examples

### Type of Change

- Adding a new entry

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=3475) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=3475) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=3475) it.</sub>

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+---
+Title: 'Functions'
+Description: 'Functions are the block of reusable code that can perform similar related actions.'
```
```suggestion
Description: 'Functions are blocks of reusable code that can be used to repeat processes throughout a script or program.'
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+Description: 'Functions are the block of reusable code that can perform similar related actions.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
```
```suggestion
```

caupolicandiaz replied:
subjects is limited to two items

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+Tags:
+  - 'Functions'
+CatalogContent:
+  - 'Computer Science'
+  - 'rust-for-programmers'
```
```suggestion
  - 'rust-for-programmers'
  - 'paths/computer-science'
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+  - 'rust-for-programmers'
+---
+
+**Rust functions** are declared using the `fn` keyword. Its arguments are type annotated, just like variables, and, if the function returns a value, the return type must be specified after an arrow `->`.
```
The entry should start with an expanded version of the description used in the metadata above. Then there should be a ## Syntax header. All of the content on lines 15 and 16 can follow the header. Generally, we also include a pseudo code block with the general syntax laid out.

It may be helpful to review an existing concept entry-
https://www.codecademy.com/resources/docs/cpp/functions

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+---
+
+**Rust functions** are declared using the `fn` keyword. Its arguments are type annotated, just like variables, and, if the function returns a value, the return type must be specified after an arrow `->`.
+Rust code uses _snake_case_ as the conventional style for function and variable names, in which all letters are lowercase and underscores separate words.
```
```suggestion
Rust code uses the _snake_case_  convention for function and variable names, in which all letters are lowercase and underscores separate words.
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,58 @@
+
+**Rust functions** are declared using the `fn` keyword. Its arguments are type annotated, just like variables, and, if the function returns a value, the return type must be specified after an arrow `->`.
+Rust code uses _snake_case_ as the conventional style for function and variable names, in which all letters are lowercase and underscores separate words.
+
```
Generally, there should be a couple of sections all separated by a header that will highlight a particular aspect of functions with this language, such as return values, best practices, etc

caupolicandiaz reviewed (changes requested):
@vrun1208 I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

vrun1208 commented:
> @vrun1208 I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

Hello @caupolicandiaz , I've updated the changes as you suggested. Kindly review the changes in the latest commit!

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,150 @@
+## Functions with return values
+
+Functions can return values to the code that calls them. Its arguments are type annotated, just like variables, and, if the function returns a value, the return type must be specified after an arrow `->`. In Rust, the return value of the function is determined by the value of the final expression in the block of the body of a function.
+Here's an example defining funtion with return values,
```
```suggestion

Here's an example defining function with return values:
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,150 @@
+
+There is no function calls, macros, or even let statements in the function `number`, just the number 5 itself, And that’s a perfectly valid function in Rust.
+
+Now, look at another example-
```
```suggestion
Now, look at another example:
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
````diff
@@ -0,0 +1,150 @@
+}
+```
+
+The output the above code will print `The value of x is: 6`. But if we place a _semicolon_ at the end of the line containing `x + 1`, changing it from an _expression_ to a _statement_, we’ll get an error!
````
```suggestion
The output the above code will print `The value of x is: 6`. However, if  a _semicolon_ at is placed at the end of the line containing `x + 1`, it will switch from an _expression_ to a _statement_, and will result in an error!
```

caupolicandiaz commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,150 @@
+
+The output the above code will print `The value of x is: 6`. But if we place a _semicolon_ at the end of the line containing `x + 1`, changing it from an _expression_ to a _statement_, we’ll get an error!
+
+Note that, Expressions do not include ending semicolons. If you add a semicolon to the end of an expression, you turn it into a statement, and it will then not return a value.
```
```suggestion
Note that expressions do not include ending semicolons. Adding a semicolon to the end of an expression turns it into a statement, and it will not return a value.
```

caupolicandiaz reviewed (changes requested):
@vrun1208 I've added additional comments. Please update, and tag me when you've completed the changes. Thanks

vrun1208 commented:
I have made changes as you suggested, @caupolicandiaz !!

caupolicandiaz reviewed (approved):
@vrun1208 I've made some direct edits to update the spacing and some other elements. This is ready for a second review.

SSwiniarski commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,154 @@
+
+## Syntax
+
+Functions in Rust are declared using the `fn` keyword. Every Rust program has at least one function: the _main_ function, which is the entry point of many programs.
```
```suggestion
Functions in Rust are declared using the `fn` keyword. Every Rust program has at least one function: the `main` function, which is the entry point of many programs.
```

SSwiniarski commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,154 @@
+
+Note that, `another_function` can also be declared before or after the function `main`. Rust doesn't care about where it is defined, it just needs to be somewhere in the scope.
+
+## Functions as Parameters
```
```suggestion
## Function Parameters
```

SSwiniarski commented on content/rust/concepts/functions/functions.md:
```diff
@@ -0,0 +1,154 @@
+
+When parameters are passed by reference, unlike value parameters, a new storage location is not created for these parameters. The reference parameters represent same memory location as the actual parameters. Parameter values can be passed by reference by prefixing the variable name with an `&`.
+
+In the example given below, there is variable `number`, which is initially `5`. A _reference_ to the variable `number` is passed to the `make_zero()` function. The function now operates on the original variable. After the function is called it returns back to main method, the value of the original variable will be zero.
```
```suggestion
In the example given below, there is variable `number`, which is initially `5`. A **reference** to the variable `number` is passed to the `make_zero()` function. The function now operates on the original variable. After the function is called it returns back to main method, the value of the original variable will be zero.
```

SSwiniarski reviewed (changes requested):
@vrun1208 I did a second review and left a number of comments. Tag me when they're addressed and I'll review again.

vrun1208 commented:
> @vrun1208 I did a second review and left a number of comments. Tag me when they're addressed and I'll review again.

Hi @SSwiniarski, I have made the changes as you suggested!

SSwiniarski reviewed (approved):
@vrun1208 I made one minor edit and this looks good to merge.
